In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split

In [2]:
iris = pd.read_csv("iris_dataset.csv")

In [3]:
iris.head()

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

        target  
0  Iris-setosa  
1  Iris-setosa  
2  Iris-setosa  
3  Iris-setosa  
4  Iris-setosa

In [4]:
iris['target'].unique()

array(['Iris-setosa', 'Iris-versicolor', 'Iris-virginica'], dtype=object)

In [5]:
iris.describe(include='all')

sepal length (cm)  sepal width (cm)  petal length (cm)  \
count          150.000000        150.000000         150.000000   
unique                NaN               NaN                NaN   
top                   NaN               NaN                NaN   
freq                  NaN               NaN                NaN   
mean             5.843333          3.054000           3.758667   
std              0.828066          0.433594           1.764420   
min              4.300000          2.000000           1.000000   
25%              5.100000          2.800000           1.600000   
50%              5.800000          3.000000           4.350000   
75%              6.400000          3.300000           5.100000   
max              7.900000          4.400000           6.900000   

        petal width (cm)       target  
count         150.000000          150  
unique               NaN            3  
top                  NaN  Iris-setosa  
freq                 NaN           50  
mean            1.198667          NaN  
std             0.763161          NaN  
min             0.100000          NaN  
25%             0.300000          NaN  
50%             1.300000          NaN  
75%             1.800000          NaN  
max             2.500000          NaN

In [6]:
iris.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   target             150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [7]:
iris.isnull().sum()

sepal length (cm)    0
sepal width (cm)     0
petal length (cm)    0
petal width (cm)     0
target               0
dtype: int64

# MODEL

In [ ]:
X=iris.iloc[:,0:4].values
y=iris.iloc[:,4].values

In [ ]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
y = le.fit_transform(y)

In [ ]:
from sklearn.naive_bayes import GaussianNB

In [ ]:
#Train and Test split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0)

In [ ]:
from sklearn.metrics import precision_score, recall_score, f1_score, accuracy_score, confusion_matrix

In [ ]:
gaussian = GaussianNB()
gaussian.fit(X_train, y_train)

In [ ]:
import pickle

In [ ]:
pickle.dump(gaussian, open('model_naive_bayes.pkl', 'wb'))

new_data = [[3.0, 3.0, 3.1, 3.0]]  # Sesuaikan dengan data baru Anda

# Melakukan prediksi
prediksi_baru = gaussian.predict(new_data)
# Hasil prediksi akan dalam bentuk angka yang mewakili kelas
prediksi_baru_label = le.inverse_transform(prediksi_baru)  # Mengubah label numerik kembali ke bentuk asli

print(f'Hasil prediksi untuk data baru: {prediksi_baru_label[0]}')